### PR TITLE
Return type checking for exists methods and Limit/PageRequest and First conflict detection

### PR DESCRIPTION
**Pull request by njr-11**
* Validate return types for exists methods, and add tests
* Reject findFirst methods that have a Limit or PageRequest parameter, and add tests
* Detect misordered special parameters on Query methods, and add tests

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by njr-11** [comment]
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=61888c89-b7c2-46dd-a853-1375789177ca) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

File: dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops [hunk 1]
```diff
@@ -980,3 +981,11 @@ CWWKD1098.spec.param.position.err.explanation=When special Jakarta Data defined
  positioned after all the query parameters.
 CWWKD1098.spec.param.position.err.useraction=Move the special Jakarta Data defined \
  parameter types to positions at the end of the repository method signature.
+
+CWWKD1099.first.keyword.incompat=CWWKD1099E: The {0} method of the {1} repository \
+ has a name that includes the First keyword, which is incompatible with the \
+ {2} parameter of the method.
+CWWKD1099.first.keyword.incompat.explanation=The First keyword cannot be used \
```
Comment (ramkumar-k-9286) [inline dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops]: CWWKD1099.first.keyword.incompat.explanation=A Query by Method Name method that includes a Limit or PageRequest parameter cannot use the First keyword.

Comment (njr-11) [reply dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops]: @ramkumar-k-9286 Thanks! I added your update under commit https://github.com/OpenLiberty/open-liberty/pull/30413/commits/e3de7abbd3ccf9b3a5053df4166d0ce27145ec62
Please review again and add your approval.

**Comment by njr-11** [comment]
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f556dc45-cd0d-4ea2-8320-7716dbf428af) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

**Review by jmstephensgit — APPROVED** [review]

**Comment by njr-11** [comment]
The first build was successful and had all tests passing, but an update was needed to a translatable message, so a second build was submitted.
The second build had 2 intermittent test failures in unrelated test buckets that do not use Jakarta Data or its updated translatable message.

**Pull request merged**